public/app/core/crash/index.ts

```diff
@@ -39,8 +45,17 @@ export function initializeCrashDetection() {
       return new Worker(new URL('./client.worker', import.meta.url));
     },
 
-    createDetectorWorker(): SharedWorker {
-      return new SharedWorker(new URL('./detector.worker', import.meta.url));
+    /**
+     *  There are limitations that require us to manually assert the type here.
+     *  1) Webpack uses static code analysis to create a new entry point for a SharedWorker.
+     *     It requies constructing an object with exact syntax new SharedWorker(...) (https://webpack.js.org/guides/web-workers/)
+     *  2) Some browsers may not support SharedWorkers hence we cannot extends CorsSharedWorker like CorsWorker and
+     *     window.SharedWorker needs to be referenced during runtime only if it supported (https://developer.mozilla.org/en-US/docs/Web/API/SharedWorker)
+     *
+     *  We guarantee the type assertion is correct by returning a SharedWorker in CorsSharedWorker constructor.
+     */
+    createDetectorWorker() {
+      return new SharedWorker(new URL('./detector.worker', import.meta.url)) as globalThis.SharedWorker;
```

Comment:
Btw, is there any reason we rename the import `CorsSharedWorker as SharedWorker`?

Instead of doing

```suggestion
      return new CorsSharedWorker(new URL('./detector.worker', import.meta.url)) as SharedWorker;
```

I think that might be a bit more readable. Otherwise it might be unclear why we need to assert a `SharedWorker` as `SharedWorker`.

Reply:
The only reason is that webpack uses static code analysis to create entry points for workers. So it requires that exact class name to be used. We do the same thing with CorsWorker (WebWorker).

Reply:
Ah, okay, that's wild.